integration_linux_test.go

```diff
@@ -0,0 +1,256 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Copyright 2018 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+//
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cni
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"sync"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	baseNetNSDir = "/var/run/netns/"
+
+	cniBridgePluginCfg = `
+{
+  "cniVersion": "1.0.0",
+  "name": "gocni-test",
+  "plugins": [
+    {
+      "type":"bridge",
+      "bridge":"gocni-test0",
+      "isGateway":true,
+      "ipMasq":true,
+      "promiscMode":true,
+      "ipam":{
+        "type":"host-local",
+        "ranges":[
+          [{
+            "subnet":"10.88.0.0/16"
+          }],
+          [{
+            "subnet":"2001:4860:4860::/64"
+          }]
+        ],
+        "routes":[
+          {"dst":"0.0.0.0/0"},
+          {"dst":"::/0"}
+        ]
+       }
+    },
+    {
+      "type":"portmap",
+      "capabilities":{
+        "portMappings":true
+      }
+    }
+  ]
+}
+`
+)
+
+// TestBasicSetupAndRemove tests the cni.Setup/Remove with real bridge and
+// loopback CNI plugins.
+//
+// NOTE:
+//
+// 1. It required that the both bridge and loopback CNI plugins are installed
```

Comment:
t.Skip when the requirement is not satisfied

Reply:
It should still be skippable. Just skipping based on a missing requirement isn't the best approach since an error in setup could just end up causing a skip. I might suggest either skipping on [`testing.Short`](https://pkg.go.dev/testing#Short) or putting the integration test in a sub-package